phoenicis-repository/src/main/java/org/phoenicis/repository/dto/ApplicationDTO.java

```diff
@@ -72,6 +69,27 @@ public String getDescription() {
         return scripts;
     }
 
+    /**
+     * Returns the main miniature belonging to this {@link ApplicationDTO}.
+     * The main miniature is the miniature with the file name <code>main.png</code>.
+     * If no such miniature exists the first miniature of this {@link ApplicationDTO} is returned.
+     * If this {@link ApplicationDTO} contains no miniatures {@link Optional#empty()} is returned.
+     *
+     * @return An optional with the found URI, or {@link Optional#empty()} if no miniature exists
+     */
+    public Optional<URI> getMainMiniature() {
+        Optional<URI> result = this.miniatures.stream().filter(uri -> uri.getPath().endsWith("main.png")).findFirst();
+
+        /*
+         * Fallback in case no main miniature has been selected but the list contains at least one other miniature
+         */
+        if (!result.isPresent()) {
+            result = this.miniatures.stream().findFirst();
```

Comment:
I'm not sure if we need this fallback, because if no `main.png` file is found we would use a random miniature that may or may not be useful for identifying the application. 
I guess in this case it may be more useful to show a generic fallback image, which we show if `Optional.empty()` is returned.
What do you think?